### DISCUSSION TITLE
Missed readings with "Brain State: Waiting Connect" / "status 133 (GATT_ERROR)" despite a scan result

### DISCUSSION THREAD

**Discussion by loopy321** [NightscoutFoundation/xDrip#2043] [Q&A]
Are there any troubleshooting tips for a "Brain State: Waiting Connect" timeout with the logs showing "**Got scan result match: Dexcom B7 D3:B8:... rssi: -7**1", "status 133 (GATT_ERROR)"? 

I do get readings, intermittently, however misses for more than 20 min at a time.

(this is on a Samsung S7 active, Android 8.0, nightly 2022.03.02, tried all of the recommended settings at https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html - I have verified that my SW50 gets readings solidly every 5 min on "slot 2" when the phone is set to "slot 1" and missing readings)

**Comment by Navid200** [comment]
Can you post a screenshot of the system status page?  You can blackout the serial number if you like.

**Comment by loopy321** [comment]
Sure:
![Screenshot_20220309-162928_xDrip+](https://user-images.githubusercontent.com/1026345/157551151-aa76b01a-3c0e-418a-b557-f7aa59859f08.jpg)

**Reply by Navid200** [reply]
That is a very low voltage B.
Can you go to logs, enable all categories and search for volt?
What do you see?

**Reply by loopy321** [reply]
It is a rebatteried 81XXXX transmitter - in my experience this is not low for Voltage B (my last one I used up until Voltage B was much lower).  Searching for "volt" in logs shows "Saving battery data: Status OK ..."

**Reply by Navid200** [reply]
Can you show a screenshot of the search results?

**Reply by loopy321** [reply]
Maybe this helps (missed reading, top of log):
![image](https://user-images.githubusercontent.com/1026345/157554093-f15ef9c8-1055-4dbc-8a76-0e3078d54b4b.png)

**Reply by Navid200** [reply]
Can you show the search results?

**Reply by loopy321** [reply]
Sure:
![image](https://user-images.githubusercontent.com/1026345/157555865-7bc3972d-168d-47b1-93b2-f49506af5c8c.png)

**Comment by Navid200** [comment]
Have you disabled battery optimization and Bluetooth battery optimization etc.?

**Reply by loopy321** [reply]
Yes, Under Settings->Battery->Power saving mode=OFF and Xdrip+ was added to the "Unmonitored Apps"
Note that there is no "Bluetooth battery optimization" setting I can alter when showing system apps - under Bluetooth, "Battery" is greyed out.  Just as in your G6-reccommended settings doc.  Thanks for that!

**Comment by Navid200** [comment]
What is new?  What changed that might have triggered this?
Did you just start using that phone?
Did you just start using this particular transmitter?
... ?

**Reply by loopy321** [reply]
This is my "backup" phone (which I recall used to work stably without the Native Algo) and I wanted to see if I could use it with the new slot settings (so upgraded to the newest Xdrip+).  It is a "new" transmitter, but I have confirmed that it gets >95% readings both with my SW50 (sony smartwatch 3 running Xdrip+) and even better when using Spike (iOS-14.4).  I suspect there is just something buggy with the Bluetooth stack and was hoping to figure it out or a workaround, thus my initial post asking for debugging tips.   
But maybe it has to do with my setting the "slot" to 1.  That said, this problem also appears even when the "slot" is set to the default=2.  

Perhaps it is just time to retire this old phone and upgrade to one that has better bluetooth software in the OS (I doubt there are many with this S7-G891A running Oreo-8.0).  I was just curious if there were any starting points for figuring out why Xdrip scans and finds the transmitter, but then fails to get readings.  

Thanks for the help!
